core/Models/Model.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace app\core\Models;
+
+use app\core\Logs;
+use app\core\Request;
+use app\core\Validate;
+
+class Model
+{
+    public array $errors = [];
+    public array $files = [];
+    public $success;
+    public $size;
+    public $type;
+    public $name;
+    public $tmpName;
+    private Logs $log;
+
+    public function __construct()
+    {
+        $this->log = new Logs();
+    }
+
+    public function uploadFile($data)
+    {
+        $this->files = $this->getFiles();
+        if (Request::isPost()) {
+            $canLoad = $this->loadData($data['file']);
+            if ($canLoad !== true) {
+                return $this;
+            }
+
+            $uploadDir = $_SERVER['DOCUMENT_ROOT'] . "/uploads";
+            @mkdir($uploadDir, 0755);
+            move_uploaded_file($this->tmpName, $uploadDir . '/' .  $this->name);
```

Comment:
Обычно имя генерируется, а не используется tmpName. При генерации имени следует учитывать следующее
````
A well-known security best practice is to never trust the input provided by users. This also applies to the files uploaded by your visitors. The UploadedFile class provides methods to get the original file extension ([getClientOriginalExtension()](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#method_getClientOriginalExtension)), the original file size ([getSize()](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#method_getSize)) and the original file name ([getClientOriginalName()](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#method_getClientOriginalName)). However, they are considered not safe because a malicious user could tamper that information. That's why it's always better to generate a unique name and use the [guessExtension()](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#method_guessExtension) method to let Symfony guess the right extension according to the file MIME type;
```

Выдержка из Symfony Doc, но  основная идея должна быть понятна